pkg/services/secrets/database/database.go

```diff
@@ -11,17 +11,27 @@ import (
 	"github.com/grafana/grafana/pkg/services/secrets"
 )
 
-const dataKeysTable = "data_keys"
-
 type SecretsStoreImpl struct {
-	db  db.DB
-	log log.Logger
+	db    db.DB
+	log   log.Logger
+	table string
 }
 
 func ProvideSecretsStore(db db.DB) *SecretsStoreImpl {
 	store := &SecretsStoreImpl{
-		db:  db,
-		log: log.New("secrets.store"),
+		db:    db,
+		log:   log.New("secrets.store"),
+		table: "data_keys",
+	}
+
+	return store
+}
+
+func ProvideSecretsStoreForTable(db db.DB, table string) *SecretsStoreImpl {
```

Comment:
Nit: I think it's better to name this something like:
```suggestion
func NewSecretsStoreForTable(db db.DB, table string) *SecretsStoreImpl {
```

Instead as `Provide*` is generally for all Wire things, and here you can just re-use this function from `ProvideSecretsStore`.